src/KeyringController.ts

```diff
@@ -826,69 +1004,37 @@ class KeyringController extends EventEmitter {
    * Deallocates all currently managed keyrings and accounts.
    * Used before initializing a new vault.
    */
-
-  /* eslint-disable require-await */
-  async clearKeyrings() {
+  async #clearKeyrings(): Promise<void> {
     // clear keyrings from memory
     this.keyrings = [];
     this.memStore.updateState({
       keyrings: [],
     });
   }
 
-  /**
-   * Update memStore Keyrings
-   *
-   * Updates the in-memory keyrings, without persisting.
-   */
-  async _updateMemStoreKeyrings() {
-    const keyrings = await Promise.all(
-      this.keyrings.map(this.displayForKeyring),
-    );
-    return this.memStore.updateState({ keyrings });
-  }
-
   /**
    * Unlock Keyrings
    *
    * Unlocks the keyrings.
    *
    * @fires KeyringController#unlock
    */
-  setUnlocked() {
+  #setUnlocked(): void {
     this.memStore.updateState({ isUnlocked: true });
     this.emit('unlock');
   }
 
-  /**
-   * Forget hardware keyring.
-   *
-   * Forget hardware and update memorized state.
-   *
-   * @param {Keyring} keyring - The keyring to forget.
-   */
-  forgetKeyring(keyring) {
-    if (keyring.forgetDevice) {
-      keyring.forgetDevice();
-      this.persistAllKeyrings();
-    } else {
-      throw new Error(
-        `KeyringController - keyring does not have method "forgetDevice", keyring type: ${keyring.type}`,
-      );
-    }
-  }
-
   /**
    * Instantiate, initialize and return a new keyring
    *
    * The keyring instantiated is of the given `type`.
    *
-   * @param {string} type - The type of keyring to add.
-   * @param {object} data - The data to restore a previously serialized keyring.
-   * @returns {Promise<Keyring>} The new keyring.
+   * @param type - The type of keyring to add.
+   * @param data - The data to restore a previously serialized keyring.
+   * @returns The new keyring.
    */
-  async _newKeyring(type, data) {
-    const keyringBuilder = this.getKeyringBuilderForType(type);
+  async #newKeyring(type: string, data: any): Promise<Keyring<State> | void> {
```

Comment:
It seems that the argument for `Keyring.deserialize` is of type `State`. Can we use that here instead of `any`?